### PR TITLE
Add promotion

### PR DESCRIPTION
**Pull request by KailiDing1121**
- Get request body in `services.py`
- Add promotion through deserialization and serialization
- Add bad data format handler
- Add database validation error handler

### CODE REVIEW

**Comment by iamnwi** [comment]
Did you write test cases?

File: service/models.py [hunk 1]
```diff
@@ -71,10 +76,31 @@ def serialize(self):
...
+        Deserializes a Promotion from a dictionary
+
+        Args:
+            data (dict): A dictionary containing the Promotion data
+        """
+        try:
+            self.code = data['code']
+            self.percentage = data['percentage']
+            self.expiry_date = datetime.fromtimestamp(int(data['expiry_date']))
+            self.start_date = datetime.fromtimestamp(int(data['start_date']))
```
Comment (iamnwi) [inline service/models.py]: how about the `products` field?

File: service/models.py [hunk 2]
```diff
@@ -71,10 +76,31 @@ def serialize(self):
             "id": str(self.id),
             "code": self.code,
             "products": self.products,
+            "percentage": self.percentage,
```
Comment (iamnwi) [inline service/models.py]: 👍

File: service/service.py [hunk 3]
```diff
@@ -68,6 +68,39 @@ def list_promotions():
...
+######################################################################
+# ADD PROMOTIONS
+######################################################################
+@app.route('/promotions', methods=['POST'])
+def add_promotions():
+    app.logger.info('Request to create a promotion')
+    check_content_type('application/json')
+    promotion = Promotion()
+    promotion.deserialize(request.get_json())
+    promotion.save()
```
Comment (iamnwi) [inline service/service.py]: Maybe we should check if this promotion has overlap in time with other promotions having the same promotion code. Because we shouldn't have two promotion with the same code that active in the same period(or their active periods have overlap)

File: service/models.py (hunk 1 above)
Comment (KailiDing1121) [reply service/models.py]: I don't know how to add and test with the `products` field since we just have an empty `Product` table. Do we need to first set up a fake `Product` table?
